Casks/qt-creator.rb

```diff
@@ -1,13 +1,22 @@
 cask "qt-creator" do
-  version "4.14.1"
-  sha256 "7fb68dff6ed1fd3b02bd82857e9394afa240fb89d2bdfe8a672335319d9906e6"
+  version "4.14.2"
+  sha256 "1a2b1c5a85f253c78085914094e6163e6b6630bdf0b3ad1b6ad6631466c57410"
 
   url "https://download.qt.io/official_releases/qtcreator/#{version.major_minor}/#{version}/qt-creator-opensource-mac-x86_64-#{version}.dmg"
-  appcast "https://download.qt.io/official_releases/qtcreator/",
-          must_contain: version.major_minor
   name "Qt Creator"
+  desc "IDE for application development"
   homepage "https://www.qt.io/developers/"
 
+  livecheck do
+    url "https://download.qt.io/official_releases/qtcreator/"
+    strategy :page_match do |page|
+      page.scan(%r{href="(\d+(?:\.\d+)*)/"}i).lazy.map do |match|
+        version_page = Net::HTTP.get(URI.parse("#{url}#{match[0]}/"))
+        version_page[%r{href="(\d+(?:\.\d+)*)/"}i, 1]
+      end.reject(&:nil?).first
+    end
+  end
```

Comment:
```suggestion
  # It's necessary to check within a major/minor version directory
  # (fetching an additional page) to obtain the full version.
  livecheck do
    url "https://download.qt.io/official_releases/qtcreator/?C=M;O=D"
    strategy :page_match do |page|
      # These version directories can sometimes be empty, so this will 
      page.scan(%r{href=["']?v?(\d+(?:\.\d+)+)/?["' >]}i).lazy.map do |match|
        version_page = Homebrew::Livecheck::Strategy.page_content(url.sub("/?", "/#{match[0]}/?"))
        next if version_page[:content].blank?

        versions = version_page[:content].scan(%r{href=["']?v?(\d+(?:\.\d+)+)/?["' >]}i).map(&:first)
        next if versions.blank?

        versions
      end.reject(&:nil?).first
    end
  end
```

Here's a suggestion that synthesizes the approaches in #104889 and this PR. I won't repeat the lengthy explanation from #104889 but some of the same feedback applies here.